include/xtypes/idl/generator.hpp

```diff
@@ -260,8 +264,16 @@ inline std::string generate_union(const UnionType& type, size_t tabs = 0)
 inline std::string aliase(const DynamicType& type, const std::string& name)
 {
     std::stringstream ss;
-    ss << "typedef " << generator::type_name(type) << " ";
-    ss << name << ";" << std::endl;
+    ss << "typedef " << generator::type_name(type) << " " << name;
+    if (type.kind() == TypeKind::ARRAY_TYPE)
+    {
+        std::vector<uint32_t> array_dims = static_cast<const ArrayType&>(type).dimensions();
```

Comment:
As you must remove the method `dimensions` from `ArrayType`, create another function here with that logic.
In lines 68 to 75 of this file, you can find a similar algorithm. That piece of code should use the new function as well.